src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/UpdateSentenceDateService.kt

```diff
@@ -151,6 +157,32 @@ class UpdateSentenceDateService(
     )
   }
 
+  private fun Licence.recordAuditEvent(auditEventDescription: String) {
```

Comment:
it's a bit weird asking for a licence entity to do something that talks to the outside world.
Entities should generally be treated as individual data objects with behaviour that modifies its own state. 
 
You could use a `with` inside a separate method to make accessing fields easier:
```kt
  with(licence) {
auditEventRepository.saveAndFlush(
      AuditEvent(
        licenceId = this.id,
        username = "SYSTEM",
        fullName = "SYSTEM",
        eventType = AuditEventType.SYSTEM_EVENT,
        summary = "$auditEventDescription for ${this.forename} ${this.surname}",
        detail = "ID ${this.id} type ${this.typeCode} status ${this.statusCode} version ${this.version}",
      ),
    )
  }
```  
which would have the same affect to put the licence in scope

Reply:
Done